freeipa/src/main/java/com/sequenceiq/freeipa/service/upgrade/UpgradeImageService.java

```diff
@@ -35,15 +50,65 @@ public ImageInfoResponse selectImage(Stack stack, ImageSettingsRequest imageSett
         return imageInfoResponse;
     }
 
-    public ImageInfoResponse currentImage(Stack stack) {
+    public ImageInfoResponse fetchCurrentImage(Stack stack) {
         ImageEntity imageEntity = imageService.getByStackId(stack.getId());
         ImageInfoResponse imageInfoResponse = new ImageInfoResponse();
         imageInfoResponse.setImageName(imageEntity.getImageName());
         imageInfoResponse.setCatalog(imageEntity.getImageCatalogUrl());
         imageInfoResponse.setCatalogName(imageEntity.getImageCatalogName());
         imageInfoResponse.setOs(imageEntity.getOs());
         imageInfoResponse.setId(imageEntity.getImageId());
+        imageInfoResponse.setDate(imageEntity.getDate());
         LOGGER.info("Current image: {}", imageInfoResponse);
         return imageInfoResponse;
     }
+
+    public List<ImageInfoResponse> findTargetImages(Stack stack, ImageSettingsRequest imageSettingsRequest, ImageInfoResponse currentImage) {
+        Optional<String> currentImageDate = getCurrentImageDate(stack, currentImage);
+        List<Pair<ImageWrapper, String>> imagesWrapperAndName = imageService.fetchImagesWrapperAndName(stack, imageSettingsRequest);
+        return imagesWrapperAndName.stream()
+                .filter(imageWrapperAndName -> currentImageDate.isPresent()
+                        && isCandidateImageNewerThanCurrent(imageWrapperAndName.getLeft(), currentImageDate.get()))
+                .filter(imageWrapperAndName -> !currentImage.getId().equals(imageWrapperAndName.getLeft().getImage().getUuid()))
```

Comment:
As far as I can see isCandidateImageNewerThanCurrent makes it sure that the candidate is different from the current image. Is this filtering here to make it sure it wont return the current one?

Reply:
you are right, it's to make sure we don't include it. theoretically it's not necessary

Reply:
thanks